contracts/cw1-subkeys/src/contract.rs

```diff
@@ -145,7 +172,7 @@ where
         if let Some(exp) = expires {
             allowance.expires = exp;
         }
-        allowance.balance = allowance.balance.sub(amount.clone())?; // Fails if no tokens
+        allowance.balance = allowance.balance.sub_saturating(amount.clone())?; // Fails if no tokens
```

Comment:
The comment "fails if no tokens". Is that out of date now?

Reply:
Right, thanks.

Reply:
Correction: this still fails if no tokens. It silently ignores underflows, but doesn't ignore a missing balance entry for that denomination. I think this is an important distinction: an admin could make a spelling mistake when trying to remove an entry, and if this fails silently, he will never be notified of the error.

Nevertheless, if you think this is not desirable / important, I'll change it to always failing silently.

Reply:
Related: Shouldn't `denom` be case insensitive, for robustness and simplicity?

Reply:
Very good distinction to make (underflow vs. no denom) and I agree with it. Nice thinking.

The comment could be a little more verbose and capture that.

Reply:
As to case insensitive... it is case sensitive in the sdk. we could normalize both stored values and the ones we match to always be lower case before comparing. I guess that would be nice. But I'm not sure if there are issues there. Can you make an issue to add that, and we can add that later, once this is merged. And then we can maybe get some comments on the idea. I am not completely sure about that one.

Reply:
Me neither... I'll create the issue.